modules/sfp_grep_app.py

```diff
@@ -0,0 +1,210 @@
+# -*- coding: utf-8 -*-
+# -------------------------------------------------------------------------------
+# Name:        sfp_grep_app
+# Purpose:     Searches grep.app API for domains, URLs and emails related to the
+#              specified domain.
+#
+# Author:      <bcoles[at]gmail[.]com>
+#
+# Created:     2020-04-12
+# Copyright:   (c) bcoles 2020
+# Licence:     GPL
+# -------------------------------------------------------------------------------
+
+import json
+import math
+import time
+import re
+import urllib.request, urllib.parse, urllib.error
+from sflib import SpiderFoot, SpiderFootPlugin, SpiderFootEvent
+
+class sfp_grep_app(SpiderFootPlugin):
+    """grep.app:Footprint,Investigate,Passive:Search Engines::Search grep.app API for links and emails related to the specified domain."""
+
+    # Default options
+    opts = {
+        'max_pages': 20,
+        'dns_resolve': True,
+    }
+
+    # Option descriptions
+    optdescs = {
+        'max_pages': "Maximum number of pages of results to fetch.",
+        'dns_resolve': "DNS resolve each identified domain."
+    }
+
+    results = None
+    errorState = False
+
+    def setup(self, sfc, userOpts=dict()):
+        self.sf = sfc
+        self.__dataSource__ = "grep.app"
+        self.results = self.tempStorage()
+        self.errorState = False
+
+        for opt in list(userOpts.keys()):
+            self.opts[opt] = userOpts[opt]
+
+    def watchedEvents(self):
+        return ["DOMAIN_NAME"]
+
+    def producedEvents(self):
+        return ["EMAILADDR", "DOMAIN_NAME", "INTERNET_NAME", "INTERNET_NAME_UNRESOLVED", "LINKED_URL_INTERNAL"]
```

Comment:
Listening for `DOMAIN_NAME` and producing it, so probably worth having a check in `handleEvent` to ensure we aren't recursive?

Reply:
The `DOMAIN_NAME` should be in `self.results` after the first run, preventing recursion.